source/client/components/CVLightsTask.ts

```diff
@@ -0,0 +1,140 @@
+import CVDocumentProvider from "./CVDocumentProvider";
+import CVNode from "./CVNode";
+import CVTask, { types } from "./CVTask";
+import { CLight, ELightType, ICVLight } from "./lights/CVLight";
+
+export default class CVLightsTask extends CVTask {
+    static readonly typeName: string = "CVLightsTask";
+
+    static readonly text: string = "Lights";
+    static readonly icon: string = "bulb";
+
+    protected static readonly ins = {
+        create: types.Event("Light.Create"),
+        delete: types.Event("Light.Delete"),
+        name: types.String("Light.Name", ""),
+        type: types.Enum("Light.Type", ELightType, ELightType.directional),
+    };
+    protected static readonly outs = {
+    };
+
+    ins = this.addInputs<CVTask, typeof CVLightsTask.ins>(CVLightsTask.ins);
+    outs = this.addOutputs<CVTask, typeof CVLightsTask.outs>(CVLightsTask.outs);
+
+    create(): void {
+        super.create();
+        this.startObserving();
+    }
+
+    dispose(): void {
+        this.stopObserving();
+        super.dispose();
+    }
+
+    createView() {
+        return new LightsTaskView(this);
+    }
+
+    activateTask(): void {
+        super.activateTask();
+    }
+
+    deactivateTask(): void {
+        super.deactivateTask();
+    }
+
+    update(context: IUpdateContext) {
+        const { ins } = this;
+
+        if (ins.create.changed) {
+            const mainView: MainView = document.getElementsByTagName('voyager-story')[0] as MainView;
+            const activeDoc = this.getMainComponent(CVDocumentProvider).activeComponent;
+
+            CreateLightMenu.show(mainView, activeDoc.setup.language)
+                .then(([selectedType, name]) => {
+                    this.createLightNode(selectedType, name);
+                    return true;
+                }).catch(e => console.error("Error creating light:", e));
+        }
+
+        const activeNode: NVNode | undefined = this.nodeProvider.activeNode;
+        if (activeNode?.light) {
+            if (ins.name.changed && activeNode.name !== ins.name.value) {
+                activeNode.name = ins.name.value;
+                return true;
+            }
+            if (ins.delete.changed) {
+                activeNode.dispose();
+                return true;
+            }
+
+            const lightType: string = ELightType[(activeNode.light.constructor as any).type];
+            if (ins.type.changed && ins.type.value !== lightType as unknown) {
+                const newNode: NVNode = this.createLightNode(ins.type.value, activeNode.name);
+                CVLightsTask.copyLightProperties(activeNode, newNode);
+                activeNode.dispose();
+                this.nodeProvider.activeNode = newNode;
+                return true;
+            }
+        }
+        return false;
+    }
+
+    protected createLightNode(newType: ELightType, newName: string): NVNode {
+        const lightType = lightTypes.find(lt => lt.type === ELightType[newType].toString());
+
```

Comment:
It could handle errors in case there are no lights with:
```
if (!lightType) {
      throw new Error(`Unsupported light type: ${newType}`);
  }
  ```